lib/Doctrine/Common/Cache/PhpFileCache.php

```diff
@@ -108,8 +108,7 @@ private function includeFileForId($id)
     {
         $fileName = $this->getFilename($id);
 
-        // note: error suppression is still faster than `file_exists`, `is_file` and `is_readable`
-        $value = @include $fileName;
+        $value = file_exists($fileName) ? @include $fileName : null;
```

Comment:
Please run benchmarks against something like phpbench/phpbench, if you want
some accurate results

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

On 26 May 2016 at 01:23, Jean Carlo Machado <notifications@github.com>
wrote:

> In lib/Doctrine/Common/Cache/PhpFileCache.php
> <https://github.com/doctrine/cache/pull/152#discussion_r64670063>:
>
> > @@ -108,8 +108,7 @@ private function includeFileForId($id)
> >      {
> >          $fileName = $this->getFilename($id);
> >
> > -        // note: error suppression is still faster than `file_exists`, `is_file` and `is_readable`
> > -        $value = @include $fileName;
> > +        $value = file_exists($fileName) ? @include $fileName : null;
>
> It seems that dealing with error_handlers is more expensive than doing a
> file_exists.
> https://3v4l.org/gOXjR
> Either way, I'm not sure this kind of micro-optimization is healthy to the
> code.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/doctrine/cache/pull/152/files/7381dfecae5864a951d171829e468fade2ef6c99#r64670063>
>